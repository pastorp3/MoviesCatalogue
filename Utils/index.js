import { checkPropTypes } from 'prop-types';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from '../src/reducers/index';

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return component;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
}
