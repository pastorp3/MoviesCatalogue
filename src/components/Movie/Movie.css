.Movie-card {
  padding: 2%;
  flex-grow: 1;
  flex-basis: 16%;
  display: flex;
  flex-direction: column;
  text-align: center;
}

.poster {
  max-width: 180px;
  max-height: 225px;
  border-radius: 10px;
  -webkit-box-shadow: -2px 6px 23px 3px rgba(0, 0, 0, 0.4);
  box-shadow: -2px 6px 23px 3px rgba(0, 0, 0, 0.4);
}

.movie-title {
  font-weight: bold;
}

.movie-info {
  padding: 10px 10px 0 10px;
}

.movie-date {
  color: gray;
}
