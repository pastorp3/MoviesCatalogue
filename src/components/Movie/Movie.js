/* eslint-disable react/forbid-prop-types */
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Movie.css';

const Movie = ({
  title,
  year,
  poster,
  id,
}) => {
  const imgSrc = `https://image.tmdb.org/t/p/w500/${poster}`;
  const link = `/movie/${id}`;
  return (
    <div data-test="Movie-card" className="Movie-card">
      <div data-test="Movie-poster">
        <Link to={link}>
          <img alt="poster" src={imgSrc} className="poster" />
        </Link>
      </div>
      <div data-info="Movie-info" className="movie-info">
        <span className="movie-title">{title}</span>
        <br />
        <span className="movie-date">{year}</span>
      </div>
    </div>
  );
};

Movie.propTypes = {
  title: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};

export default Movie;

/* eslint-enable */
