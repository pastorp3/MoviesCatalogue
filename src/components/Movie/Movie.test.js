import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Movie from './Movie';
import { checkProps, findByTestAtrr } from '../../../Utils';


const setUp = (props={
  title: 'Test Title',
  year: '01-01-2021',
  poster: 'Test Poster',
  id: 1,
}) => {
  const component = shallow(<Movie {...props} />);
  return component;
}

describe('Movie Component', () => {
  describe('Checking PropTypes', () => {
    it('Should not throw a warning', () => {
      const expectedProps = {
        title: 'Test Title',
        year: '01-01-2021',
        poster: 'Test Poster',
        id: 1,
      };
      const propsErr = checkProps(Movie, expectedProps);
      expect(propsErr).toBeUndefined();
    })
  });

  let component;
  beforeEach(() => {
    component = setUp();
  })

  it('Should render whithout error', () => {
    const wrapper = findByTestAtrr(component, 'Movie-card');
    expect(wrapper.length).toBe(1);
  });

  it('Should render movie poster whitout error', () => {
    const wrapper = findByTestAtrr(component, 'Movie-poster');
    expect(wrapper.length).toBe(1);
  });

  it('Should render movie infor whithout error', () => {
    const wrapper = findByTestAtrr(component, 'Movie-info');
    expect(wrapper.length).toBe(1);
  })

  it('Should render according with snapshout whithout error', () => {
    const props = {
      title: 'Test Title',
      year: '01-01-2021',
      poster: 'Test Poster',
      id: 1,
    }
    const tree = renderer.create(<BrowserRouter><Movie {...props} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})