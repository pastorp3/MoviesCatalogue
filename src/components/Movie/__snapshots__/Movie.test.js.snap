// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Component Should render according with snapshout whithout error 1`] = `
<div
  className="Movie-card"
  data-test="Movie-card"
>
  <div
    data-test="Movie-poster"
  >
    <a
      href="/movie/1"
      onClick={[Function]}
    >
      <img
        alt="poster"
        className="poster"
        src="https://image.tmdb.org/t/p/w500/Test Poster"
      />
    </a>
  </div>
  <div
    className="movie-info"
    data-info="Movie-info"
  >
    <span
      className="movie-title"
    >
      Test Title
    </span>
    <br />
    <span
      className="movie-date"
    >
      01-01-2021
    </span>
  </div>
</div>
`;
